### PR TITLE
implemented console log screenshot

### PR DESCRIPTION
**Pull request by JGarcia-li**

### CODE REVIEW

File: src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java [hunk 1]
```diff
@@ -13,20 +13,23 @@
...
 import java.io.Writer;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
 
 @Slf4j
 public class Screenshot {
 
     private static final String screenShotDirectory = getDirectory("screenshots");
     private static final String htmlScreenShotDirectory = getDirectory("html-screenshots");
+    private static final String consoleLogDirectory = getDirectory("console-log-screenshots");
```
Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java]: it's not really a screenshot so I'd name it better

File: src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java [hunk 2]
```diff
@@ -13,20 +13,23 @@
...
 
     public static void takeScreenshot(String filename) {
         if (log.isDebugEnabled()) {
             takeFullDesktopScreenshot(filename);
         } else {
             if (DriverManager.INSTANCE.isDriverStarted()) {
                 try {
-                    filename +=  "_" + System.currentTimeMillis() + "_" + Thread.currentThread().getId() + ".png";
+                    filename +=  "_" + Thread.currentThread().getId() + ".png";
```
Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java]: This won't work. There are times where the filename is the same (different runs through the same method) and the thread could be the same. Also multiple runs could stomp on each other unless you clean. I'd keep the TS

File: src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java [hunk 3]
```diff
@@ -80,6 +83,37 @@ public static void takeHTMLScreenshot(String filename) {
         }
     }
 
+    public static void takeConsoleLogScreenshot(String filename){
```
Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java]: change the method name to something like saveConsoleLog or more descriptive

File: src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java [hunk 4]
```diff
@@ -80,6 +83,37 @@ public static void takeHTMLScreenshot(String filename) {
...
+                                               .filter(Level.SEVERE)
+                                               .stream()
+                                               .map(logEntry -> logEntry.getMessage().concat("\n"))
+                                               .collect(Collectors.joining()));
+        } catch (IOException ex) {
+            log.info("Unable to write out current state of the console log");
+        } finally {
+            try {
+                writer.close();
+            } catch (Exception e) {
```
Comment (jchang691) [inline src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java]: Should this be IOException?

Comment (jchang691) [inline src/main/java/com/lithium/mineraloil/selenium/elements/Screenshot.java]: Should this be IOException?

**Pull request merged**